x-pack/plugins/apm/public/context/url_params_context/helpers.test.ts

```diff
@@ -134,8 +157,35 @@ describe('url_params_context helpers', () => {
         ).toEqual({
           start: '1970-01-01T00:00:00.000Z',
           end: '1970-01-01T00:00:00.000Z',
+          exactStart: '1970-01-01T00:00:00.000Z',
+          exactEnd: '1970-01-01T00:00:00.000Z',
         });
       });
     });
   });
+
+  describe('getExactDate', () => {
+    it('returns undefined when date in not provided', () => {
+      expect(helpers.getExactDate()).toBeUndefined();
+    });
+    it('returns undefined when date in not relative', () => {
+      expect(helpers.getExactDate('2021-01-28T05:47:52.134Z')).toBeUndefined();
+    });
+    it('returns exact date', () => {
+      jest
+        .spyOn(datemath, 'parse')
+        .mockReturnValue(moment('2021-06-02T17:56:49.260Z').utc());
+      expect(helpers.getExactDate('now-24h/h')?.toISOString()).toEqual(
+        '2021-06-02T17:56:49.260Z'
+      );
+    });
+    it('returns original date when now/d is passed', () => {
+      jest
+        .spyOn(datemath, 'parse')
+        .mockReturnValue(moment('2021-06-02T17:00:00.000Z').utc());
+      expect(helpers.getExactDate('now/d')?.toISOString()).toEqual(
+        '2021-06-02T17:00:00.000Z'
```

Comment:
How does this test verify that the date is not being rounded when the original date is already rounded.

Reply:
yeah, it actually doesn't make much sense to expect for a specific date. I'd rather spy the `datemath.parse` and check if it was called correctly.